test/shmemx/shmem_malloc_with_hints.c

```diff
@@ -0,0 +1,101 @@
+/*
```

Comment:
A test on this API is being added in both `unit` and `shmemx` dir. Not sure if we wanted to keep both of them.

Reply:
I think it's ok - the `shmemx` version tests the "NO_BARRIER" hint that @kholland-intel was looking into.  Maybe we can rename the test in v1.5.3 to distinguish it.